src/metabase/upload.clj

```diff
@@ -281,16 +282,19 @@
       (rows->schema header (sample-rows rows)))))
 
 (defn load-from-csv
-  "Loads a table from a CSV file. If the table already exists, it will throw an error. Returns nil."
-  [driver db-id table-name csv-file]
+  "Loads a table from a CSV file. If the table already exists, it will throw an error. Returns nil. Deletes the file
+  after loading."
+  [driver db-id table-name ^File csv-file]
   (let [col->upload-type   (detect-schema csv-file)
         col->database-type (update-vals col->upload-type (partial driver/upload-type->database-type driver))
         column-names       (keys col->upload-type)]
     (driver/create-table driver db-id table-name col->database-type)
     (try
       (let [rows (parsed-rows col->upload-type csv-file)]
-        (driver/insert-into driver db-id table-name column-names rows))
+        (driver/insert-into driver db-id table-name column-names rows)
+        (.delete csv-file))
```

Comment:
Do we need to delete the temp files created by multipart params here? 

It should be handled by the middleware to delete in an hour:
https://github.com/ring-clojure/ring/blob/master/ring-core/src/ring/middleware/multipart_params/temp_file.clj#L51

Reply:
Not strictly necessary, but seems like a "better safe than sorry" situation since you we have very finite disk space on AWS and an hour's worth of 200 MB files adds up

Reply:
Fair enough, good thinking